protocols/netty/src/main/java/org/apache/james/protocols/netty/AbstractAsyncServer.java

```diff
@@ -128,17 +125,19 @@ protected void configureBootstrap(ServerBootstrap bootstrap) {
     
     @Override
     public synchronized void unbind() {
-        if (started == false) {
+        if (!started) {
             return;
         }
 
+        List<Future<?>> futures = new ArrayList<>();
         if (bossGroup != null) {
-            bossGroup.shutdownGracefully();
+            futures.add(bossGroup.shutdownGracefully());
         }
 
         if (workerGroup != null) {
-            workerGroup.shutdownGracefully();
+            futures.add(workerGroup.shutdownGracefully());
         }
+        futures.forEach(Throwing.<Future<?>>consumer(Future::await).sneakyThrow());
```

Comment:
I have some fixups : gracefull shutdown take time and we might not want it for tests 

CF https://github.com/apache/james-project/pull/898

IMO graceful shutdown is not a requirement for Netty 4 adoption but merely a nice enhancement. I would likely not add it in this pull request and keep it separate...

Reply:
commit removed